### PR TITLE
bug 1801938: add host-etcd-2 which will never list the bootstrap etcd as an endpoi…

### PR DESCRIPTION
**Pull request by deads2k**
…nt and never have valid hostnames

This eliminates the possibility of dns dependence downstream. It requires an update to the installer to pass the bootstrap IP to be a complete replacement, but they have to have the ability to pass the arg first.

### CODE REVIEW

File: pkg/operator/hostendpointscontroller2/host_endpoints_controller.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+		endpointAddresses = append(endpointAddresses, corev1.EndpointAddress{
+			IP:       nodeInternalIP,
+			NodeName: &node.Name,
+		})
+	}
+
+	// sort to keep a stable order
+	sort.Slice(endpointAddresses, func(i, j int) bool {
+		return (endpointAddresses)[i].Hostname < (endpointAddresses)[j].Hostname
```
Comment (sanchezl) [inline pkg/operator/hostendpointscontroller2/host_endpoints_controller.go]: no `Hostname` to sort by

File: pkg/operator/hostendpointscontroller2/host_endpoints_controller.go [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+						Protocol: "TCP",
+					},
+				},
+			},
+		},
+	}
+}
+
+func (c *HostEndpoints2Controller) applyEndpoints(required *corev1.Endpoints) error {
+	existing, err := c.endpointsLister.Endpoints(operatorclient.TargetNamespace).Get("host-etcd")
```
Comment (sanchezl) [inline pkg/operator/hostendpointscontroller2/host_endpoints_controller.go]:
```suggestion
	existing, err := c.endpointsLister.Endpoints(operatorclient.TargetNamespace).Get("host-etcd-2")
```

**Comment by deads2k** [comment]
/hold

this must be hitting some conditoin that didn't exist when it was first written.

**Comment by deads2k** [comment]
/hold cancel

I see it.  Weird that cluster-bootstrap message sucked.

**Pull request merged**